addons/mail/wizard/mail_compose_message.py

```diff
@@ -1127,8 +1126,20 @@ def _process_mail_values_state(self, mail_values_dict):
             elif not mail_to_normalized or not email_re.findall(mail_to):
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_email_invalid'
-            elif done_emails is not None and not mailing_document_based:
-                done_emails.append(mail_to)
+            elif mail_to in sent_emails_mapping:
+                # if using exact copies of attachments, mark duplicate if contents are the same as any other mail
+                if len(self.attachment_ids) == len(mail_values.get('attachment_ids', [])) and\
+                   any(sent_mail.get('subject') == mail_values.get('subject') and
+                       sent_mail.get('body') == mail_values.get('body') for sent_mail in sent_emails_mapping[mail_to]):
+                    mail_values['state'] = 'cancel'
+                    mail_values['failure_type'] = 'mail_dup'
+                    mail_values['failure_reason'] = _('This email was already sent to this address')
```

Comment:
Description of "failure_reason" says:
> Failure reason. This is usually the exception thrown by the email server, stored to ease the debugging of mailing issues.

Here, I don't feel like it adds anything, failure type already gives this info, it will just bloat the database.

Reply:
(This is marked as resolved but still there? Maybe you did not push yet)

Reply:
Mm weird, removed (again)